### ISSUE TITLE
Failing spack build due to issue in Albany_STKDiscretization.cpp

### ISSUE THREAD

**Issue by ikalash** [sandialabs/Albany#879]
The nightly spack build is failing due to the following error: 
```
420    /tmp/ikalash/spack-stage/spack-stage-albany-develop-jfp3ybpbdshnwqb
            uck3xssyfjmdmjwmq/spack-src/src/disc/stk/Albany_STKDiscretization.c
            pp: In member function 'void Albany::STKDiscretization::setupExodus
            Output()':
  >> 421    /tmp/ikalash/spack-stage/spack-stage-albany-develop-jfp3ybpbdshnwqb
            uck3xssyfjmdmjwmq/spack-src/src/disc/stk/Albany_STKDiscretization.c
            pp:2311:63: error: invalid conversion from 'Albany_MPI_Comm' {aka '
            int'} to 'stk::ParallelMachine' {aka 'ompi_communicator_t*'} [-fper
            missive]
     422     2311 |         new stk::io::StkMeshIoBroker(getMpiCommFromTeuchosC
            omm(comm)));
     423          |                                      ~~~~~~~~~~~~~~~~~~~~~~
            ~~~^~~~~~
```
I suspect it's from the PR merge that was done on 12/22 by @bartgol .  Could you please have a look @bartgol (or anyone else)?

It's likely the issue will show up in other nightlies.  They just happened to all have failed last night due to some Trilinos failures.  I'm fixing it now, so we should know tomorrow.

**Comment by ikalash** [comment]
I just confirmed that this shows up in a regular Albany build w/o spack: https://sems-cdash-son.sandia.gov/cdash/viewBuildError.php?buildid=43483 .

**Comment by ikalash** [comment]
So, it actually looks like this is due to Trilinos, not Albany.  I attempted a fix and am not getting undefined reference errors to utilities in Albany_CommUtils.hpp: https://sems-cdash-son.sandia.gov/cdash/viewBuildError.php?buildid=43484 .  I verified that if I check out master Trilinos instead of develop, the problem goes away.  After the break, it would be great if someone (@mperego ?, @bartgol ?, @jewatkins ?) can look into this and get the issue resolved.  I sadly do not have the bandwidth right now.  I will probably switch the nightlies to use master Trilinos for the time being so that they are not failing.

As an aside, I don't know whose bright idea it was to merge in a bunch of changes to Trilinos that literally broke everything on multiple levels the day before the shutdown...

**Comment by ikalash** [comment]
Probably my fix https://github.com/sandialabs/Albany/commit/97930f4e5e8c52844a691a298ce497b8ed4b86d9 can/should be removed after understanding why there are problems now with the Albany_CommUtilities.cpp and coming up with a real fix.

**Comment by jewatkins** [comment]
Just a wild guess but maybe the tribits snapshot is clashing with our cmake config. Last passing commit ids I see is Albany: 9c32d4722, Trilinos: 0270d8bd4cd so it looks like it was after some of the epetra depr. went in.

**Comment by jewatkins** [comment]
Agree, not the best thing to happen right before the shutdown... I think I recall a similar situation happened a few years ago...

**Comment by ikalash** [comment]
@jewatkins : I agree that the tribits clashing is likely the culprit, esp. given that they just pushed a bunch of changes related to tribits.  I feel like we should triage it a bit on the Albany side before we open an issue.  If you'd like to volunteer for this (after the break), please speak up.

**Comment by jewatkins** [comment]
I haven't really been following the changes that have been going into Trilinos/Albany but I can try a build after the break if no one else volunteers and the builds are still broken.

**Comment by bartlettroscoe** [comment]
>  I agree that the tribits clashing is likely the culprit,

@ikalash, what do you mean by "TriBITS crashing"?  Do you mean the HDF5 disable bringing down the configure https://github.com/trilinos/Trilinos/issues/11426?

**Comment by ikalash** [comment]
I'm not sure what exactly is going on, quite honestly.  We made no changes to Albany but are getting link errors if we build against Trilinos develop now.  It doesn't happen when I build against Trilinos master.  @lxmota is finding that when he builds LCM against a Trilinos build with MPI enabled, Albany says there was no MPI in Trilinos.  Perhaps you have ideas on how your changes could lead to this behavior?  I think realistically we will not get the chance to troubleshoot this until after the shutdown on the Albany side.

**Comment by bartlettroscoe** [comment]
> I think realistically we will not get the chance to troubleshoot this until after the shutdown on the Albany side.

@ikalash, okay, let me just put in a PR to back out the Trilinos PR https://github.com/trilinos/Trilinos/pull/11380 for now.  I will just need to get someone to approve it.

**Comment by bartlettroscoe** [comment]
@ikalash, I just created https://github.com/trilinos/Trilinos/pull/11428 to revert the changes.  I just need a Trilinos GitHub project member to approve the PR and it will merge after PR testing completes.

**Comment by ikalash** [comment]
Thanks!  I'm happy to approve it, if you add me to the list of possible approvers.  It doesn't seem I can approve it when I click on it right now.
